### ISSUE TITLE
RADICAL_PILOT_DBURL becomes mandatory?

### ISSUE THREAD

**Issue by lee212** [radical-cybertools/radical.pilot#2006]
It seems default_dburl is not offered when `RADICAL_PILOT_DBURL` is missing.
```
Traceback (most recent call last):
  File "/gpfs/alpine/scratch/hrlee/csc393/radical.pilot.sandbox/rp.session.login
4.hrlee.018241.0017/pilot.0000/rp_install/bin/radical-pilot-agent-statepush", li
ne 67, in <module>
    mongo, db, _, _, _ = ru.mongodb_connect(str(dburl))
  File "/gpfs/alpine/scratch/hrlee/csc393/radical.pilot.sandbox/rp.session.login
4.hrlee.018241.0017/pilot.0000/rp_install/lib/python3.7/site-packages/radical/ut
ils/misc.py", line 99, in mongodb_connect
    user, pwd,    ssl] = split_dburl(dburl, default_dburl)
  File "/gpfs/alpine/scratch/hrlee/csc393/radical.pilot.sandbox/rp.session.login
4.hrlee.018241.0017/pilot.0000/rp_install/lib/python3.7/site-packages/radical/ut
ils/misc.py", line 37, in split_dburl
    raise ValueError("expected 'mongodb[+ssl]://' url, not '%s'" % dburl)
ValueError: expected 'mongodb[+ssl]://' url, not 'None'
```

**Comment by andre-merzky** [comment]
This seems to be the agent side, and on the agent, a default URL would not work (it needs to use the same URL as the client side).  What code did you run, on what stack and resource?

**Comment by lee212** [comment]
```
 python               : 3.7.0
  pythonpath           : /sw/summit/xalt/1.1.4/site:/sw/summit/xalt/1.1.4/libexec
  virtualenv           : /ccs/home/hrlee/venv3/entk.copy2
  radical.entk         : 0.90.6-v0.72.0-81-gad4d968@fix-issue_393
  radical.pilot        : 0.90.6-v0.90.1-157-g4f05a4f@devel
  radical.saga         : 0.90.6-bv0.72.1-67-g5deb253@devel
  radical.utils        : 0.90.6-v0.90.3-168-gd27dc3a@devel
```
This is Summit

**Comment by andre-merzky** [comment]
So what did you run?  Or are you saying that summit is broken in general for RP?  Can you reproduce this with the RP  examples (w/o EnTK), and if so, can you provide some instructions on how to reproduce it?

**Comment by lee212** [comment]
This is generated when `00_getting_started.py` executed.

**Comment by mturilli** [comment]
@lee212 , following what we discussed a couple of weeks ago on #devel, does this refer to the fact that when a mongodb endpoint is not specified, RP returns that trace instead of a meaningful exception? More in general, is this about the expectation that if a mongodb endpoint is not passed to RP, RP should use a default value? If so, I would not assume a default value but I would assume a meaningful exception for the user.

**Comment by lee212** [comment]
RP on client side is okay, https://github.com/radical-cybertools/radical.pilot/blob/f0a66dc41e062053e177fddaae292c4ef437e844/src/radical/pilot/configs/session_default.json#L6 is provided but the error message in this issue is generated during the bootstrap which means STDOUT/ERR on the client won't provide meaningful exception messages but failure.

**Comment by mturilli** [comment]
@andre-merzky do we really have an hard coded login/psswd in our config file!? ;) 
@lee212 thank you for clarifying this, I had not understood the problem indeed.

**Comment by lee212** [comment]
Just for the record, STDOUT on the client side is:
```
(entk.copy3) [hrlee@login4.summit user_guide]$ env|grep DBURL
(entk.copy3) [hrlee@login4.summit user_guide]$
(entk.copy3) [hrlee@login4.summit user_guide]$ python get_started.py.bak
EnTK session: re.session.login4.hrlee.018246.0007
Creating AppManagerSetting up RabbitMQ system                                 ok
                                                                              ok
Validating and assigning resource manager                                     ok
Setting up RabbitMQ system                                                   n/a
new session: [re.session.login4.hrlee.018246.0007]                             \
database   : [mongodb://rct:rct_test@two.radical-project.org/rct_test]        ok
create pilot manager                                                          ok
submit 1 pilot(s)
        [ornl.summit:1]
                                                                              ok
wait for 1 pilot(s)
                                                                              ok
closing session re.session.login4.hrlee.018246.0007                            \
close pilot manager                                                            \
wait for 1 pilot(s)
                                                                         timeout
                                                                              ok
session lifetime: 131.5s                                                      ok
All components terminated
Traceback (most recent call last):
  File "get_started.py.bak", line 38, in <module>
    appman.run()
  File "/ccs/home/hrlee/venv3/entk.copy3/lib/python3.7/site-packages/radical/entk/appman/appmanager.py", line 385, in ru                                                                                                                                                                                                     n
    'ended in state %s' % res_alloc_state)
radical.entk.exceptions.EnTKError: Cannot proceed. Resource ended in state DONE
```

**Comment by andre-merzky** [comment]
Thanks for reporting this - this took me a while to understand, but I got it now.  This should be fixed in PR #2013.

**Comment by andre-merzky** [comment]
> @andre-merzky do we really have an hard coded login/passwd in our config file!? ;)

Well, yes, because how could we otherwise make it the default? :-)  But that's the DB we wanted to purge every night.  I am actually not sure if we ever set up that cron job - did we?
